scripts/build-catalogue

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+import subprocess as sp
+import sys
+from tempfile import TemporaryDirectory
+import os
+from pathlib import Path
+import shutil
+import stat
+import string
+import argparse
+
+def parse_arguments():
+    def append_slash(s):
+        return s+'/' if s and not s.endswith('/') else s
+
+    class ConciseHelpFormatter(argparse.HelpFormatter):
+        def __init__(self, **kwargs):
+            super(ConciseHelpFormatter, self).__init__(max_help_position=20, **kwargs)
+
+        def _format_action_invocation(self, action):
+            if not action.option_strings:
+                return super(ConciseHelpFormatter, self)._format_action_invocation(action)
+            else:
+                optstr = ', '.join(action.option_strings)
+                if action.nargs==0:
+                    return optstr
+                else:
+                    return optstr+' '+self._format_args(action, action.dest.upper())
+
+    parser = argparse.ArgumentParser(
+        description = 'Generate dynamic catalogue and build it into a shared object.',
+        usage = '%(prog)s catalogue_name mod_source_dir',
+        add_help = False,
+        formatter_class = ConciseHelpFormatter)
+
+    parser.add_argument('name',
+                        metavar='name',
+                        type=str,
+                        help='Catalogue name.')
+
+    parser.add_argument('modpfx',
+                        metavar='modpfx',
+                        type=str,
+                        help='Directory name where *.mod files live.')
+
+    parser.add_argument('-v', '--verbose',
+                        action='store_true',
+                        help='Verbose.')
+
+
+    parser.add_argument('-h', '--help',
+                        action='help',
+                        help='display this help and exit')
+
+    return vars(parser.parse_args())
+
+args    = parse_arguments()
+arb     = Path(__file__).parents[1]
```

Comment:
Here is my reasoning on this: The user should have to do as few things as humanely possible, since they are probably a scientist, not a developer. That means that this script should try to infer the most likely option, which is in my opinion that this script resides in the 'correct path' (in the sense that the caller wants to use the source tree above it).

Here's my suggestion for accomodating as many people as possible:

> add the `resolve()` to guard against symlinks and add an option to override the default path to the source dir.